string node_name
---
string[] names